playground/connect-next/app/login/actions.ts

```diff
@@ -73,5 +79,65 @@ export async function postPasskeyLogin(session: string) {
   }
 }
 
-// Try to log in the user to Cognito (not implemented yet)
-export async function postConventionalLogin(email: string, password: string) {}
+function createSecretHash(username: string, clientId: string, clientSecret: string) {
+  return crypto
+    .createHmac('sha256', clientSecret)
+    .update(username + clientId)
+    .digest('base64');
+}
+
+export async function startConventionalLogin(email: string, password: string) {
+  try {
+    if (!email || !password) {
+      throw new Error('Email and password are required.');
+    }
+
+    const client = new CognitoIdentityProviderClient({
+      region: process.env.AWS_REGION!,
+      credentials: {
+        accessKeyId: process.env.AWS_ACCESS_KEY_ID!,
+        secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY!,
+      },
+    });
+
+    const command = new InitiateAuthCommand({
+      AuthFlow: 'USER_PASSWORD_AUTH',
+      ClientId: process.env.AWS_COGNITO_CLIENT_ID!,
+      AuthParameters: {
+        USERNAME: email,
+        PASSWORD: password,
+        SECRET_HASH: createSecretHash(
+          email,
+          process.env.AWS_COGNITO_CLIENT_ID!,
+          process.env.AWS_COGNITO_CLIENT_SECRET!,
+        ),
+      },
+    });
+
+    const response = await client.send(command);
+
+    if (!response.AuthenticationResult?.AccessToken) {
+      throw new Error('Authentication failed. Please check your credentials and try again.');
+    }
+
+    const decoded = await verifyToken(response.AuthenticationResult.AccessToken);
+    const username = decoded.username;
+
+    if (email) {
+      cookies().set('displayName', email);
+      cookies().set('identifier', username);
+    }
+
+    return;
+  } catch (err) {
+    if (err instanceof Error) {
```

Comment:
Error handling is not working on Vercel:
![image](https://github.com/user-attachments/assets/5649091b-c469-4ef1-9891-25ce4b3ed79d)